android/WorkshopCompleted/app/src/main/java/com/mobiledgex/workshopskeleton/MainActivity.java

```diff
@@ -609,7 +620,9 @@ protected void onPostExecute(Boolean cloudletFound) {
             if (!cloudletFound) {
                 findCloudletStatusText = "Failed to find cloudlet. " + findCloudletStatusText;
                 Log.e(TAG, findCloudletStatusText);
-                showErrorMsg(findCloudletStatusText);
+                if (!errorShown) {
+                    showErrorMsg(findCloudletStatusText);
+                }
```

Comment:
If I understood correctly this check will remove duplicate error messages, I am just concerned if the order errors appears matters. (Or if there is a datetime string next to the error) 

Also maybe QoSPosBackgroundRequest() needs the same check if it is still active in the new app version.

Reply:
There is a timestamp.

Good point about QoSPosBackgroundRequest. PQOS/PQOE isn't currently supported by any carrier, and this code should probably be disabled.

Reply:
@ah1053 Now I see that it is actually working currently, I guess using random simulator data. So I'll leave the code intact. I've added the same duplicate check to the QOS code.